packages/uploads/web/src/components/RedwoodUploadsComponent.tsx

```diff
@@ -0,0 +1,145 @@
+  rejectClassName = '',
+  buttonClassName = '',
+  showButton = false, // Optionally show a button for selecting files
+  buttonText = 'Select files',
+  rejectMessage = 'Invalid file(s). Please check requirements and try again.', // Simplified reject message
+  defaultMessage = "Drag 'n' drop some files here, or click to select files", // Custom default message
+  activeMessage = "Drag 'n' drop some files here, or click to select files", // Custom active message
+  fileRenderer: FileRendererComponent = DefaultFileRenderer, // Default to simple file renderer
+  fileRejectionRenderer:
+    FileRejectionRendererComponent = DefaultFileRejectionRenderer, // Default to simple file renderer
+  ...dropzoneOptions
+}) => {
+  const [internalAcceptedFiles, setInternalAcceptedFiles] = useState<File[]>([])
+  const [internalFileRejections, setInternalFileRejections] = useState<
+    FileRejection[]
+  >([])
+
+  const {
+    getRootProps,
+    getInputProps,
+    open, // `open` allows opening the file dialog programmatically
+    isDragActive,
+    isDragReject,
+  } = useDropzone({
+    onDrop: (acceptedFiles, fileRejections, event) => {
+      setAcceptedFiles?.(acceptedFiles)
+      setFileRejections?.(fileRejections)
+      setInternalAcceptedFiles(acceptedFiles)
+      setInternalFileRejections(fileRejections)
+      onDrop?.(acceptedFiles, fileRejections, event)
+    },
+    accept,
+    maxFiles,
+    minSize,
+    maxSize,
+    multiple: maxFiles > 1 ? true : multiple, // Set multiple to true if maxFiles > 1
+    noClick: showButton,
+    ...dropzoneOptions,
+  })
+  useEffect(() => {
+    if (acceptedFiles && acceptedFiles.length === 0) {
+      getRootProps().ref.current?.dropzone?.removeAllFiles()
+    }
+  }, [acceptedFiles, getRootProps])
+
+  useEffect(() => {
+    if (fileRejections && fileRejections.length === 0) {
+      getRootProps().ref.current?.dropzone?.removeAllFiles()
+    }
+  }, [fileRejections, getRootProps])
+
+  // Function to generate the active message
+  const getActiveMessage = (): string => {
+    if (typeof activeMessage === 'function') {
+      return activeMessage({ maxFiles, minSize, maxSize, accept })
+    }
+    return activeMessage
+  }
+
+  // Function to generate the reject message
+  const getRejectMessage = (): string => {
+    if (typeof rejectMessage === 'function') {
+      return rejectMessage({ maxFiles, minSize, maxSize, accept })
+    }
+    return rejectMessage
+  }
+
+  // Function to generate the default message
+  const getDefaultMessage = (): string => {
+    if (typeof defaultMessage === 'function') {
+      return defaultMessage({ maxFiles, minSize, maxSize, accept })
+    }
+    return defaultMessage
+  }
+
+  const combinedClassName = `${className} ${
+    isDragActive ? activeClassName : ''
+  } ${isDragReject ? rejectClassName : ''}`
+
+  return (
+    <section>
+      <div {...getRootProps({ className: combinedClassName })}>
+        <input {...getInputProps({ name })} />
+        <p>
```

Comment:
Now that I look at this, it might be better to have this render some children instead of setting messages. and then maybe can place preview renderers where one wants to and well as define message components. Would need to export out some of the isDragActive and accpectedfiles to use perhaps.